### ISSUE TITLE
Patch 3 Broke the script extender and is causing crashes

### ISSUE THREAD

**Issue by CloudberryXD** [Norbyte/bg3se#64]
I use the script extender with 5e Spells, but with the new patch that just come out, it no longer works and causes the game to crash with the message of

BG3 Script Extender Error;
Could not locate RuleAction:: Call in osiris.dll

Removing the script file and 5e spells does make the game work again however.
![image](https://github.com/Norbyte/bg3se/assets/145786507/98b69f8c-f320-4d0a-ad3a-154af2632db3)

**Comment by xRaeko98** [comment]
Have the same issue, how do i update i dont see a new release ? ty in advance

> Should be fixed now.

**Comment by kraoro** [comment]
> Should be fixed now.
Github still has the previous version from three weeks ago as the latest one, though?

**Comment by Bnidoya** [comment]
> Should be fixed now.

What confuses me is that I still haven't seen the latest version. How can I download the latest version? I see that the latest version is still V3. Please let me know, I really need it

**Comment by DonPetrol** [comment]
The mod auto updates the latest release version works (The one from 2 weeks ago).

**Comment by kraoro** [comment]
I sort of figured as much, since I didn't experience any problems with any mods using this as a dependency, but I wanted to be certain. Thanks.

**Comment by jadescxrpio** [comment]
I'm still having the same error/crashing issues, is this expected?

**Comment by Pyr0blad3** [comment]
for everyone I think v5 is the newest version currently.
so if you dont have that its not ScriptExtender that crashes your game. its an old version i would assume!

proof:
![2023-09-23 02_38_39-Window](https://github.com/Norbyte/bg3se/assets/28388430/144669b4-6222-471d-a5be-0b38be359c2e)
![2023-09-23 02_39_11-Window](https://github.com/Norbyte/bg3se/assets/28388430/fef8caa2-1051-487b-bda4-69d790147a6c)

**Comment by BelegCufea** [comment]
TLDR;
It autoupdates on game launch (you can do it through or without Larian launcher).

======
There is no new version in releases on github (as there are only updater versions ... see below).
No need to run [BG3MM](https://github.com/LaughingLeader/BG3ModManager) and do anything there. 
You just need to have latest Updater ([DWrite.dll](https://github.com/Norbyte/bg3se/releases)) in game bin folder (as for time of writing this post there is v3 there). This DLL will download/update latest [Script Exteneder](https://github.com/Norbyte/bg3se) version.
As for the script extender location and version see briliant [post](https://github.com/Norbyte/bg3se/issues/64#issuecomment-1732178067) from [Pyr0blad3](https://github.com/Pyr0blad3) above (my files are in %LocalAppdata%/BG3ScriptExtender/ScriptExtender though).

**Comment by hugplx** [comment]
Mine isn't auto updating and crashing on startup. Script Extender location only has up to 4.0.0.0, no v5

**Comment by PassiveModding** [comment]
Deleting the old ` %Appdata%/Local/BG3ScriptExtender/...` folder and reinstalling script extender worked for me. For some reason I was stuck on v2/v3

**Comment by Golconda522** [comment]
I am struggling as it keeps showing v4 and won't seem to update I see two consoles for extender, I have removed it and verified the game reinstalled and it still isn't updating and my game crashes every time don't even get a loading screen just crashes. I wish I could just download the most recent dwrite.dll as the updater doesn't seem to work for me.

**Comment by hugplx** [comment]
> Deleting the old ` %Appdata%/Local/BG3ScriptExtender/...` folder and reinstalling script extender worked for me. For some reason I was stuck on v2/v3

This worked for me, thank you. Next time I reloaded the game the V5 showed and no problem.

**Comment by Pyr0blad3** [comment]
@PassiveModding @Golconda522 @hugplx 
pay attention when ScriptExtender updates to v6, v7, v8 and so on, but you dont get the new version in BG3ScriptExtender folder, the issue still exists for you in the furture!

**Comment by belowzeros** [comment]
i experienced the script extender crash myself after patch 3.  I found my ScriptExtender directory had only v1 and v2 in it...it had never updated past that so I guess it was stuck.  i nuked the directory as the others suggested trying to get it to force update.  

so now it's returning an error panel when launching the game that it can't connect to SSL and nothing downloads.

**Comment by Pyr0blad3** [comment]
@BelegCufea @hugplx @Golconda522 @PassiveModding @BelegCufea @jadescxrpio @kraoro @DonPetrol @Bnidoya 
if any of you still have problems people try downloading my v5 files in this post: https://github.com/Norbyte/bg3se/issues/66

others already fixed their issues with just installing a "working" v5 for ScriptExtender. Please read my "long" indepth post where you can find out which files you need to replace in order to install those files.

i also just in case this autoupdate issue wont get fixed whenever ScriptExtender v6 and further come out, try to provide new copied of my version files as long as they work for me, of course!

cheers.

**Comment by belowzeros** [comment]
@Pyr0blad3 

thanks for the suggestion.  i managed to fix it by downloading the manual version myself that I found crawling through the bug posts.  There is also a direct download page for all versions for future updating that is helpful.

**Comment by Heropodium** [comment]
@Pyr0blad3 

Your v5 works for me but everytime i launch the game, the bg3 mod manager, automatically installs a previous version of v5, that makes my game crash with the osiris error. The only method i know for me, to not making that happen, is to disconnect my lan cable when i launch the game. 

Any advice to prevent that previous version to get installed without constantly disconnecting my internet?

**Comment by DemonTheDark00** [comment]
I have this problem too and I don't know how to solve it.

**Comment by Pyr0blad3** [comment]
@Heropodium sadly not, as i have no clue how the auto update works. i think it works with the DWrite.dll file in your BG3/bin folder, but yeah idk how to edit that, or how that works.

@DemonTheDark00 - try follow https://github.com/Norbyte/bg3se/issues/66#issuecomment-1732157205 and download my files and try it with them i guess.

if that doesnt work i cant really help sorry.

**Comment by Heropodium** [comment]
@Pyr0blad3 I ended up deleting and reinstalling my game. I put the v5 again with the DWrite.dll and the auto update stopped working somehow LOL. Now i'm fine.

**Comment by CodeNoxAlt** [comment]
Hey! Having the same problem with mine, keep getting the "Could not locate RuleAction:: Call in osiris.dll" error using the new V6. Auto updates to V6, tried using the V5 files given by Pyr0blad3 but doesn't seem like it will use them. Just continues to download V6 files. Used this and still wouldn't work: https://github.com/Norbyte/bg3se/issues/53
Uninstalled and reinstalled, didn't work. Nothing I do seems to work. Guess I'll just hope it's fixed in the next update, unless anyone has any other ideas. Don't think BG3MM or Vortex has anything to do with it, including the auto updating.

**Comment by DemonTheDark00** [comment]
> @Heropodium sadly not, as i have no clue how the auto update works. i think it works with the DWrite.dll file in your BG3/bin folder, but yeah idk how to edit that, or how that works.
> 
> @DemonTheDark00 - try follow [#66 (comment)](https://github.com/Norbyte/bg3se/issues/66#issuecomment-1732157205) and download my files and try it with them i guess.
> 
> if that doesnt work i cant really help sorry.

I tried it and it worked for a moment but then the game crashed and now it doesn't work anymore

**Comment by BelegCufea** [comment]
Don't know if it is related, but there seems to be some problems with Vulkan (bg3.exe) runtime and v6 version. Can you try running DX11 (bg3_dx11.exe) runtime.

**Comment by Pyr0blad3** [comment]
to do that go in BG3 Mod manager -> settings -> open preferences -> general -> enable Direct x11 mode

**Comment by Pyr0blad3** [comment]
@CodeNoxAlt  - try following this: https://github.com/Norbyte/bg3se/issues/66#issuecomment-1732157205
if it doesnt work i cant really help any further.
in that case either you are doing something wrong or you have a case where its not fixable from your side i guess.

**Comment by Pyr0blad3** [comment]
v6 is here just in case!
https://github.com/Norbyte/bg3se/issues/90#issuecomment-1746715480

**Comment by Norbyte** [comment]
I'm closing old issues related to old game versions, since those were largely fixed by later game patches and SE updates and are no longer relevant/reproducible.
